src/hooks/use-reach-errors.ts

```diff
@@ -0,0 +1,36 @@
+import { APIManager } from "@stakekit/api-hooks";
+import { signal, useSignalEffect } from "@preact/signals-react";
+import { useState } from "react";
+import { ErrorsSet, errorsSet } from "../utils/errors";
+
+export const reachErrorKey = signal<ReachError | undefined>(undefined);
+
+interface ReachError {
+  message: ErrorsSet;
+  details?: { [key: string]: any };
+}
+
+APIManager.getInstance()!.interceptors.response.use(undefined, (error) => {
+  reachErrorKey.value = error.response.data as ReachError;
```

Comment:
we should add type guard to check if this is in fact `ReactError` and not some other http error data